controllers/openstackcluster_controller.go

```
@@ -627,3 +617,77 @@ func handleUpdateOSCError(openstackCluster *infrav1.OpenStackCluster, message er
 	openstackCluster.Status.FailureReason = &err
 	openstackCluster.Status.FailureMessage = pointer.String(message.Error())
 }
+
+// validateSubnets validates if the amount of IPv4 and IPv6 subnets is allowed.
+func validateSubnets(subnets []infrav1.Subnet) error {
+	isIPv4 := false
+	isIPv6 := false
+
+	for _, subnet := range subnets {
+		ip, _, err := net.ParseCIDR(subnet.CIDR)
+		if err != nil {
+			return err
+		}
+		netIP := net.ParseIP(ip.String())
+		if netIP != nil {
+			if netIP.To4() != nil {
+				if !isIPv4 {
+					isIPv4 = true
+				} else {
+					return fmt.Errorf("multiple IPv4 Subnet not allowed on OpenStackCluster")
+				}
+			} else {
+				if !isIPv6 {
+					isIPv6 = true
+				} else {
+					return fmt.Errorf("multiple IPv6 Subnet not allowed on OpenStackCluster")
+				}
+			}
+		}
+	}
```

we probably want to return an error if `netIP` is nil, right?